internal/pkg/service/common/etcdop/op/util.go

```diff
@@ -0,0 +1,55 @@
+package op
+
+import (
+	"bytes"
+	"sort"
+)
+
+func removeOpsOverlaps(ops []TrackedOp) (out []TrackedOp) {
+	// Order operations with the smallest key and with the largest range to begin
+	sort.SliceStable(ops, func(i, j int) bool {
+		// Sort by operation
+		a, b := ops[i], ops[j]
+		if a.Type != b.Type {
+			return a.Type < b.Type
+		}
+
+		// Lower key first, if the key is the same, then longer range fist
+		start := bytes.Compare(a.Key, b.Key)
+		end := bytes.Compare(a.RangeEnd, b.RangeEnd)
+		return start == -1 || start == 0 && end == 1
+	})
+
+	// Skip operations that are already covered by some previous operation
+skip:
+	for _, op := range ops {
+		for _, compareWith := range out {
+			if isOpSubsetOf(op, compareWith) {
+				// The op is already covered with the compareWith operation, skip
+				continue skip
+			}
+		}
+		out = append(out, op)
+	}
```

Comment:
Potom nam to staci iba postupne iterovat, a kedze tie najvacsie `ranges`, ktore mozu pokryvat ostatne su na zaciatku.

Pri kazdej `op` skontrolujeme, ze uz nebola pokryta niektorou predchadzajucou.